tests/smoke/Makefile

```diff
@@ -1,4 +1,12 @@
-test:
+SHELL:=/bin/bash
+
+deps:
+	git clone https://github.com/sstephenson/bats.git
+	mkdir ~/bin || :
```

Comment:
Since the nuances are going to change based on whatever the CI engine is, perhaps we have a pattern
where any command prefixed with `travis-` does the extra work required to getting it running in Travis CI:
```
make deps   =   get/install dependencies locally
make travis-deps  = get/install dependencies for travis ci
make smoketests = run smoketests locally, assuming you have all deps in place
make travis-smoketests = get/install deps for smoketests, run them, cleanup afterwards
```